modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2333,6 +2376,41 @@ def broadcast_apply_full_axis(
             else new_axis
             for i, new_axis in enumerate([new_index, new_columns])
         ]
+        # we can compute lengths and widths when `keep_partitioning=True` and `new_axes` is not none
+        new_row_lengths = None
+        new_column_widths = None
+        if self._row_lengths_cache is not None and self._column_widths_cache is not None:
+            from modin.core.storage_formats.pandas.utils import compute_chunksize
+            new_row_lengths = self._row_lengths_cache
+            new_column_widths = self._column_widths_cache
+            if axis == 0:
+                if sum(new_column_widths) != len(new_axes[1]):
+                    # previous cache isn't valid
+                    new_column_widths = None
+                if not (sum(new_row_lengths) == len(new_axes[0])):
```

Comment:
```suggestion
                if (sum(new_row_lengths) != len(new_axes[0])):
```
to be consistent with condition above 😄